### ISSUE TITLE
Getting AWS account ID for integration

### ISSUE THREAD

**Issue by sarneaud** [splunk-terraform/terraform-provider-signalfx#118]
Thanks for the provider.  I'm using it to connect SignalFX to an AWS account.

An AWS integration using roles needs a bare minimum of three resources:

* A `signalfx_aws_external_integration`
* An `aws_iam_role`
* A `signalfx_aws_integration`

Trouble is, `signalfx_aws_integration` depends on the AWS IAM role, but the IAM role needs to be set up with an assume-role policy that allows access from the SignalFX AWS account.  So Terraform needs the account number when creating the role, but it's not in the `signalfx_aws_external_integration` state.

I tried just using the 134183635603 number used in https://github.com/terraform-providers/terraform-provider-signalfx/issues/111 and other examples, but Terraform failed with "Unrecognized service response for the dry-run request" when creating the `signalfx_aws_integration` (I made sure I was using the role ARN, not ID.)

Then I got the correct AWS account ID manually from the SignalFX web dashboard, updated the `aws_iam_role` config, and then reran Terraform to reconfigure the role and finish the AWS integration.  (Actually, the first rerun gave me the same error, but a second one worked.)

It would be great if AWS account ID could be retrieved automatically in Terraform.  If the `signalfx_aws_external_integration` resource exposed it, that would be perfect.

Versions:
```
$ terraform --version
Terraform v0.12.16
+ provider.aws v2.40.0
+ provider.signalfx v4.10.1
```

Sample config:
```
resource "signalfx_aws_external_integration" "foo" {
  name = "Foo AWS"
}

resource "signalfx_aws_integration" "foo" {
  enabled            = true
  import_cloud_watch = true
  enable_aws_usage   = false

  integration_id = signalfx_aws_external_integration.foo.id
  external_id    = signalfx_aws_external_integration.foo.external_id
  role_arn       = aws_iam_role.signalfx.arn

  poll_rate = 300
      
  regions = ["us-east-1"]
}     
    
resource "aws_iam_role" "signalfx" {
  name               = "signalfx"
  description        = "SignalFX access to AWS account"
  assume_role_policy = data.aws_iam_policy_document.signalfx_assume_role.json
} 
  
data "aws_iam_policy_document" "signalfx_assume_role" {
  statement {
    actions = ["sts:AssumeRole"]

    # Need account ID here
    principals {
      type = "AWS"
      identifiers = ["arn:aws:iam::134183635603:root"]
    }

    condition {
      test     = "StringEquals"
      variable = "sts:ExternalId"
      values   = [signalfx_aws_external_integration.foo.external_id]
    }
  }
}

# Snip aws_iam_role_policy that grants lots of read access
```

**Comment by cory-signalfx** [comment]
Hey @sarneaud, thanks for filing this. Another user mentioned this to me in a VC call the other day, so I'll be taking a look.

**Comment by cory-signalfx** [comment]
Ok, I did some sleuthing and this is **not** a value that is returned when the provider makes the initial integration creation call. This is a set of values that are configured for SignalFx's various realms and I have no access to it from the provider.

I could hardcode the values or add them to docs, but for now I think the best is to leave the documentation as is that asks users to ask support for the appropriate account id.

In the meantime I've filed an internal request to add this to the integration create call.
